tests/unit/test_slide.py

```diff
@@ -159,15 +183,39 @@ def it_knows_its_dimensions(self, tmpdir):
 
         assert slide_dims == (500, 500)
 
-    def it_raises_error_if_bad_args_for_extract_tile(self, tmpdir):
-        slide, _ = base_test_slide(
-            tmpdir, PILIMG.RGBA_COLOR_500X500_155_249_240, use_largeimage=True
+    def it_extracts_tile_with_mpp_without_image_resizing(self, tmpdir, request):
+        slide = Slide(SVS.CMU_1_SMALL_REGION, tmpdir, use_largeimage=True)
+        property_mock(request, PIL.Image.Image, "size", return_value=(128, 128))
```

Comment:
shouldn't we assert that the mocks are called? (comment is referring also to other tests)

Reply:
not for properties cause we are actually using a real slide

Reply:
mmm, so what? 
![](https://media3.giphy.com/media/xT0xeuOy2Fcl9vDGiA/giphy-downsized-medium.gif)

Reply:
honestly this `extract_tile` is becoming huge. The more i see it the more I'd like to refactor it. So for now i would keep this test as is, it's kinda of weird but it's the only thing I can imagine today. Let's discuss further in the next future. What do u think?

Reply:
yes agreed, it's huge but let's merge as it is